### PR TITLE
FIX: improve VoiceOver accessibility for controls

### PR DESCRIPTION
**Pull request by 5w4kjd95yk-eng**
## Summary

- Hide decorative icon glyphs from the VoiceOver accessibility tree.
- Add explicit labels and state-aware names to icon-only and icon-plus-text controls across wallet creation, settings, send/receive, Lightning, camera, entropy, and wallet export flows.
- Preserve existing visual appearance.

## Validation

- npm test
- 53 unit suites passed (515 passed, 1 skipped)
- 14 integration suites passed; 1 environment-dependent suite skipped (106 passed, 23 skipped)
- Local iPhone build/deploy succeeded previously with bundle ID com.monicawegner.bluewallet.dev

### CODE REVIEW

File: screen/receive/AztecoRedeem.tsx [hunk 1]
```diff
@@ -103,7 +103,12 @@ const AztecoRedeem = () => {
         </BlueText>
 
         <View style={styles.selectWallet1}>
-          <TouchableOpacity accessibilityRole="button" style={styles.selectTouch} onPress={handleSelectWallet}>
+          <TouchableOpacity
+            accessibilityRole="button"
+            accessibilityLabel={loc.wallets.select_wallet}
```
Comment (GladosBlueWallet) [inline screen/receive/AztecoRedeem.tsx]: **[HIGH]** VoiceOver hears *Select wallet* while the button reads *Redeem to wallet*. Congratulations — you've built a lie detector for blind users.

File: components/Tabs.tsx [hunk 2]
```diff
@@ -32,7 +32,14 @@ export const Tabs: React.FC<TabsProps> = ({ active, onSwitch, tabs, isIpad = fal
   return (
     <View style={[tabsStyles.root, isIpad && tabsStyles.marginBottom]} onLayout={onLayout}>
       {tabs.map((Tab, i) => (
-        <TouchableOpacity key={i} accessibilityRole="button" testID={`Tab${i}`} onPress={() => onSwitch(i)} style={tabsStyles.tabRoot}>
+        <TouchableOpacity
+          key={i}
+          accessibilityRole="button"
```
Comment (GladosBlueWallet) [inline components/Tabs.tsx:38]: **[HIGH]** You stamp `accessibilityState.selected` on tabs but never name them. Combined with your `Icon` purge, ProvideEntropy's icon-only tabs become mute buttons. Selected, but unknowable. Poetic, useless.

File: screen/settings/ElectrumSettings.tsx [hunk 3]
```diff
@@ -401,7 +401,13 @@ const ElectrumSettings: React.FC = () => {
   }, [config?.connected, config?.host, config.port, createServerAction, host, isPreferred, serverHistory]);
 
   const HeaderRight = useMemo(
-    () => <HeaderMenuButton actions={generateToolTipActions()} onPressMenuItem={onPressMenuItem} />,
+    () => (
+      <HeaderMenuButton
+        actions={generateToolTipActions()}
+        onPressMenuItem={onPressMenuItem}
+        accessibilityLabel={loc.settings.electrum_preferred_server}
```
Comment (GladosBlueWallet) [inline screen/settings/ElectrumSettings.tsx]: **[MEDIUM]** You labeled a multi-action server menu *Preferred Server*. It opens history, suggestions, and reset — not a shrine to one server. Misleading labels are still lies, just subtler.

File: screen/send/Confirm.tsx [hunk 4]
```diff
@@ -339,6 +339,8 @@ const Confirm: React.FC = () => {
                 <Text style={styles.payjoinText}>Payjoin</Text>
                 <Switch
                   testID="PayjoinSwitch"
+                  accessibilityLabel="Payjoin"
```
Comment (GladosBlueWallet) [inline screen/send/Confirm.tsx]: **[MEDIUM]** Inline `Payjoin` in `accessibilityLabel` — CONTRIBUTING says strings live in `loc/en.json`. Even switches deserve localization; this one doesn't get a pass.

File: components/Icon.tsx [hunk 5]
```diff
@@ -72,6 +72,7 @@ const Icon = <T extends IconType = 'font-awesome'>({
   const isFa6 = type === 'font-awesome-6';
   const fa6IconStyle = isFa6 ? (typeof iconStyle === 'string' ? iconStyle : 'solid') : undefined;
   const mergedStyle = isFa6 ? style : [style, iconStyle];
+  const iconAccessibilityLabel = onPress ? undefined : accessibilityLabel;
```
Comment (GladosBlueWallet) [inline components/Icon.tsx:85]: **[MEDIUM]** App-wide `Icon` behavior changed — decorative glyphs now hidden by default — yet zero tests. CONTRIBUTING asked for tests when they make sense. Apparently they don't, to you.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
VoiceOver accessibility pass across 34 files — mostly `accessibilityLabel`/`accessibilityState` plumbing plus global `Icon` hiding. Intent is clear (FIX, not feature). **AztecoRedeem mislabels a redeem control**, **ProvideEntropy icon-only tabs go silent** after `Icon` hides unlabeled glyphs, and a few labels violate project conventions. Not the worst test subject I've seen — just don't get comfortable.

### Inline findings (could not anchor on diff)

- `screen/wallets/ProvideEntropy.tsx:257` — **[HIGH]** TollTab/D6Tab/D20Tab are bare icons with no `accessibilityLabel`. Your global `Icon` hide makes them vanish from the tree — VoiceOver gets 'button' and static. Entropy, inaudibly.

File: tests/unit/icon.test.tsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+jest.mock('@react-native-vector-icons/fontawesome', () => {
+  const ReactLocal = require('react');
+  return {
+    __esModule: true,
+    default: (props: Record<string, unknown>) => ReactLocal.createElement('Text', props),
+  };
+});
+
+describe('Icon accessibility', () => {
+  it('hides unlabeled decorative icons from accessibility', () => {
```
Comment (GladosBlueWallet) [inline tests/unit/icon.test.tsx:6]: **[HIGH]** You mocked FontAwesome into a prop echo chamber and assert the mock received your props. CONTRIBUTING calls that bullshit; so do I. Zero proof VoiceOver behavior changed.

File: tests/unit/transaction-status.test.tsx [hunk 7]
```diff
@@ -277,6 +277,16 @@ describe('TransactionStatus regression', () => {
     jest.clearAllMocks();
   });
 
+  it('exposes the block explorer control as a button', async () => {
```
Comment (GladosBlueWallet) [inline tests/unit/transaction-status.test.tsx]: **[HIGH]** Congratulations — you verified the developer typed `accessibilityRole` and `accessibilityLabel` into JSX. No press handler, no explorer navigation, no edge case. Mock cosplay, not a test.

File: screen/receive/ReceiveDetails.tsx [hunk 8]
```diff
@@ -561,13 +560,22 @@ const ReceiveDetails = () => {
             <>
               <View style={styles.cardSpacer} />
               <View style={styles.addressRow}>
+                <Text
```
Comment (GladosBlueWallet) [inline screen/receive/ReceiveDetails.tsx]: **[HIGH]** A phantom 1×1 `Text`, nine lines of `accessibilityOnly` styling, card set `accessible={false}`, and re-enabled interactive copy — Rube Goldberg for what one `accessibilityLabel` on `CopyTextToClipboard` could do. Overengineered bloat.

File: screen/wallets/ProvideEntropy.tsx [hunk 9]
```diff
@@ -200,7 +200,13 @@ const Dice = ({ push, sides }: { push: TPush; sides: number }) => {
   return (
     <ScrollView contentContainerStyle={[styles.diceContainer, stylesHook.diceContainer]}>
       {[...Array(sides)].map((_, i) => (
-        <TouchableOpacity accessibilityRole="button" testID={`Dice${sides}Roll${i + 1}`} key={i} onPress={() => push(getEntropy(i, sides))}>
+        <TouchableOpacity
+          accessibilityRole="button"
+          accessibilityLabel={String(i + 1)}
```
Comment (GladosBlueWallet) [inline screen/wallets/ProvideEntropy.tsx]: **[MEDIUM]** Dice faces announce "3" and "17" with zero context while tab labels got proper strings. VoiceOver thinks it's counting, not rolling entropy.

File: screen/transactions/TransactionStatus.tsx [hunk 10]
```diff
@@ -1316,13 +1319,19 @@ const TransactionStatus: React.FC = () => {
           <BlueText style={[styles.detailLabel, stylesHook.detailLabel]}>{loc.transactions.details_note}</BlueText>
           <View style={styles.detailValueContainer}>
             {memo ? (
-              <TouchableOpacity onPress={handleNotePress} activeOpacity={0.7} style={styles.memoContainer}>
+              <TouchableOpacity
```
Comment (GladosBlueWallet) [inline screen/transactions/TransactionStatus.tsx:1322]: **[MEDIUM]** Existing memo gets `accessibilityRole="button"` but no label or hint — screen reader reads note text, not "edit note." Add-note path has visible copy; edit path doesn't.

File: components/Icon.tsx [hunk 11]
```diff
@@ -80,14 +81,17 @@ const Icon = <T extends IconType = 'font-awesome'>({
...
+      importantForAccessibility={iconAccessibilityLabel ? 'yes' : 'no-hide-descendants'}
+      accessibilityLabel={iconAccessibilityLabel}
       testID={testID}
     />
   );
 
   if (onPress) {
     return (
-      <Pressable accessibilityRole="imagebutton" onPress={onPress} style={containerStyle}>
+      <Pressable accessibilityRole="imagebutton" accessibilityLabel={accessibilityLabel} onPress={onPress} style={containerStyle}>
```
Comment (GladosBlueWallet) [inline components/Icon.tsx:104]: **[LOW]** Interactive icon with no `accessibilityLabel` yields a silent `imagebutton` — hidden glyph, hidden purpose. No callers today; latent footgun for the next header icon.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
Accessibility fix sweep across VoiceOver labels, decorative icon hiding, and selected-state announcements — sensible intent, mostly mechanical diffs. Your new tests, however, are prop-parroting theater, and `ReceiveDetails` smuggled in a 1×1 invisible-text contraption Aperture would reject for excessive cube count. Not cake-worthy; not incinerator-worthy either.
